doc/dev/testing.md

```diff
@@ -0,0 +1,68 @@
+# Unit Testing
+
+You can run the unit tests after completing a build using the following:
+
+```bash
+cd build/release/stage0
+ctest -j 4 --output-on-failure --timeout 300

```

Comment:
There is some overlap here with `## Development Setup`. I like `make test` better because it ensures the build is up to date. Also, there should never be a need to run tests in stage 0. Not sure if we need the timeout specifier, it seems like some rare race condition in a single server test case on Windows.

Reply:
Ah, maybe not that rare... we should fix the test, though I'm not exactly sure how given the circumstances. Perhaps we should activate excessive logging for it on CI and wait for it to fail again.